frontend/ARCHITECTURE.md

```diff
@@ -0,0 +1,186 @@
+# Pages
+
+There are 5 main Pages in the application, several modals, and one page that is used for authentication.
+
+## Authentication
+
+The `LogIn` component located at `src/components/log-in/log-in.tsx` manages the authentication credentials that are received from the backend. Canvas and the backend authenticate with each other and then the backend redirects canvas to `<base url>/course/<courseId>?token=<JWT>`
+
+The `LogIn` component captures the course ID and the JWT and saves them to the application state, then redirects to the base URL.
+
+## Main Pages
+
+The main pages are:
+
+1. Home
+2. Gradebook
+3. GradebookSession
+4. Present
+5. SessionInProgress
+
+### Home
+
+#### Route
+
+`<base url>/`
+
+#### File
+
+`src/components/home-page/home.tsx`
+
+#### About
+
+The `Home` page is where the user begins. It is the question management screen and is where users go to create, edit, delete, and present questions. The user is able to navigate to the `Gradebook` page from here, as well as activate the present, create/edit question, and create folder modals.
+
+### Gradebook
+
+#### Route
+
+`<base url>/gradebook`
+
+#### File
+
+`src/components/gradebook/gradebook.tsx`
+
+#### About
+
+The `Gradebook` page is where the user views the grades of students from past sessions. The user is able to navigate to the `Home` and `GradebookSession` pages from here, as well as activate the export grades modal.
+
+### GradebookSession
+
+#### Route
+
+`<base url>/gradebook/:sessionId`
+
+#### File
+
+`src/components/session/session.tsx`
+
+#### About
+
+The `GradebookSession` page is where the user views the grades of students from a specific session. This view shows the grades on a per question basis instead of a per session basis like the main gradebook. The user is able to navigate to the `Home` and `Gradebook` pages from here, as well as activate the export grades modal.
+
+### Present
+
+#### Route
+
+`<base url>/session/present`
+
+#### File
+
+`src/components/present-session/present-session.tsx`
+
+#### About
+
+The `Present` page is where the user goes to after selecting questions to present in the present modal. This relies on a session being started in the present modal, and is where the user will wait while students join their session. There is a count of how many students are connected to the session. Once the user is ready to start displaying questions, they navigate to the `SessionInProgress` screen, which is the only navigable screen from this page.
```

Comment:
> they navigate to the `SessionInProgress` screen, which is the only navigable screen from this page

They also go back to the home page smh my smh

Reply:
Ah true. Shake smh my smh head smh